src/transformers/models/tvlt/feature_extraction_tvlt.py

```diff
@@ -0,0 +1,521 @@
+        pixel_masks = np.array(pixel_masks)
+        
+        padded_pixel_features = np.zeros([len(pixel_features), self.num_frames, 3, self.pixel_size, self.pixel_size]).astype(np.float32)
+        for i in range(len(pixel_features)):
+            pixel_feature = pixel_features[i]
+            padded_pixel_features[i, : pixel_feature.shape[0], :, :, :] = pixel_feature
+        
+        # return as BatchFeature
+        data = {"pixel_values": padded_pixel_features, "pixel_masks": pixel_masks}
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+
+        return encoded_inputs
+    
+    
+    
+class TVLTAudioFeatureExtractor(FeatureExtractionMixin, ImageFeatureExtractionMixin):
+    r"""
+    Constructs a TVLT audio feature extractor. This feature extractor can be used to prepare audios for the model.
+
+    This feature extractor inherits from [`FeatureExtractionMixin`] which contains most of the main methods. Users
+    should refer to this superclass for more information regarding those methods.
+
+    Args:
+        
+    """
+
+    model_input_names = ["audio_values", "audio_masks"]
+
+    def __init__(
+        self,
+        audio_size=1024,
+        num_channels=1,
+        audio_patch_size=[16, 16],
+        feature_size=128,
+        sampling_rate=44100,
+        hop_length=512,
+        chunk_length=30,
+        n_fft=2048,
+        padding_value=0.0,
+        return_attention_mask=False,  # pad inputs to max length with silence token (zero) and no attention mask
+        **kwargs
+    ):
+        super().__init__(
+            feature_size=feature_size,
+            sampling_rate=sampling_rate,
+            padding_value=padding_value,
+            return_attention_mask=return_attention_mask,
+            **kwargs,
+        )
+        
+        self.audio_size = audio_size
+        self.num_channels = num_channels
+        self.audio_patch_size = audio_patch_size
+        self.frequency_length = feature_size // self.audio_patch_size[1]
+        self.n_fft = n_fft
+        self.hop_length = hop_length
+        self.chunk_length = chunk_length
+        self.n_samples = chunk_length * sampling_rate
+        self.nb_max_frames = self.n_samples // hop_length
+        self.sampling_rate = sampling_rate
+        self.mel_filters = self.get_mel_filters(sampling_rate, n_fft, n_mels=feature_size)
+
+    def normalize_pixel(self, audio, mean, std):
+
+        # normalize
+        if not isinstance(mean, np.ndarray):
+            mean = np.array(mean).astype(video.dtype)
+        if not isinstance(std, np.ndarray):
+            std = np.array(std).astype(video.dtype)
+        
+        return (audio - mean[None, None, None]) / std[None, None, None]
+
+    def get_mel_filters(self, sr, n_fft, n_mels=128, dtype=np.float32):
```

Comment:
Use the `# Copied from whisper` as it seems to be the exact same code

Reply:
This needs to be `# Copied from transformers.models.whisper.feature_extraction_whisper.WhisperFeatureExtractor. get_mel_filters ` ;)

Reply:
Seems like this wasn't resolved